src/main/java/nl/tudelft/contextproject/gui/CreateScriptController.java

```diff
@@ -33,140 +49,380 @@
+     *      filled with.
      */
-    public static void setFill(boolean bFill) {
-        fill = bFill;
+    private void fillTable(List<Shot> shots) {
+        tableEvents.getItems().addAll(shots);
     }
 
     /**
-     * Allows for editable rows in the table.
+     * Allows for editable rows in the table. This method defines the
+     * necessary components for row by row editing of the table.
      */
     private void allowEditing() {
+        editBox.setVisible(false);
+        editBox.toBack();
+
+        // Allows for getting the last selected row.
+        lastSelectedRow = new SimpleObjectProperty<>();
+        tableEvents.setRowFactory(tableView -> {
+            TableRow<Shot> row = new TableRow<Shot>();
+
+            row.selectedProperty().addListener((obs, wasSelected, isNowSelected) -> {
+                if (isNowSelected) {
+                    lastSelectedRow.set(row);
+                }
+            });
+
+            return row;
+        });
+
+        editBlockScroll.setOnMouseClicked(event -> {
+            editBlockScroll.setVisible(false);
+            editConfirmAction(lastSelectedRow.get().getItem());
+        });
+
+        btnEditConfirm.setOnAction(event -> {
+            boolean emptyField = false;
+
+            if (editCamera.getSelectionModel().isEmpty()) {
+                editCamera.setStyle("-fx-border-color: red;");
+                emptyField = true;
+            }
+
+            if (editPreset.getSelectionModel().isEmpty()) {
+                editPreset.setStyle("-fx-border-color: red;");
+                emptyField = true;
+            }
+
+            if (editDescription.getText().isEmpty()) {
+                editDescription.setStyle("-fx-border-color: red;");
+                emptyField = true;
+            }
+
+            if (!emptyField) {
+                editCamera.setStyle("");
+                editPreset.setStyle("");
+                editDescription.setStyle("");
 
-        /**
-         * TODO Finish WIP code
-         * This code is being worked on but is not finished before sprint 4.
-         * 
+                editConfirmAction(lastSelectedRow.get().getItem());
+            }
+        });
+
+        btnEditRemove.setOnAction(event -> {
+            Shot shot = lastSelectedRow.get().getItem();
+            tableEvents.getItems().remove(shot);
+
+            editDoneAction();
+        });
+
+        EventHandler<KeyEvent> addResourceHandler = event -> {
+            if (lastSelectedRow.get() != null && event.getCode() == KeyCode.ENTER) {
+                editConfirmAction(lastSelectedRow.get().getItem());
+            }
+        };
+
+        editBox.setOnKeyPressed(addResourceHandler);
+        editShot.setOnKeyReleased(addResourceHandler);
+        editCamera.setOnKeyReleased(addResourceHandler);
+        editPreset.setOnKeyReleased(addResourceHandler);
+        editDescription.setOnKeyReleased(addResourceHandler);
+
+        initTable();
+    }
```

Comment:
This is a very long method. It would be nice if we could shorten it. I do not see an optimal way to do this, but maybe at least initialize the buttons in a separate method?